### ISSUE TITLE
Mask not applied to negative coordinates

### ISSUE THREAD

**Issue by ChristianKienle** [pixijs/pixijs#7952]
### Expected Behavior
Mask also applies for negative coordinates.

### Current Behavior
Mask is not applied for negative coordinates.

### Steps to Reproduce
[Open the pixi playground project I prepared](https://www.pixiplayground.com/#/edit/1HNLK9IUxinLo5Q_q3Q3f) and click on refresh. You will see something like this:

<img width="273" alt="Screenshot 2021-11-04 at 14 54 46" src="https://user-images.githubusercontent.com/153225/140326355-108c5c0f-9dca-4d15-9124-338cb2701f48.png">

What you should see is this:

<img width="287" alt="Screenshot 2021-11-04 at 14 55 00" src="https://user-images.githubusercontent.com/153225/140326391-8c57dd5f-008d-4bd2-820c-dc82e277dbea.png">

I have done the following:
- Created a subclass of `PIXI.Container` called `Bunnies`. This container simply renders 100x100 bunnies (using `PIXI.Sprite`) in a grid.
- Created an instance of `PIXI.Container` called `bunniesContainer`. 
- I then simply add an instance of `Bunnies` to the `bunniesContainer`.
- I then apply a mask on the `bunniesContainer` which should limit the drawing of the bunnies to a rect `0,0,200,200`.
- I then add the `bunniesContainer` to the stage and position it at `30,30`.
- I then position the instance of `bunnies` at `-50,-50` and the mask does no longer seem to be applied.
- However it is still applied to all positive coorinates. 

### Environment

- **`pixi.js` version**: latest stable
- **Browser & Version**: Chrome (latest stable)
- **OS & Version**: macOS
- **Running Example**: [https://www.pixiplayground.com/#/edit/1HNLK9IUxinLo5Q_q3Q3f](https://www.pixiplayground.com/#/edit/1HNLK9IUxinLo5Q_q3Q3f)

**Comment by bigtimebuddy** [comment]
I'm not sure I understand this issue, it works as I would expect. 

"mask does no longer seem to be applied" => but your content is masked in your preview image

Try adding your mask to the stage to visualize what's going on:
https://www.pixiplayground.com/#/edit/r2XABvAo-3xGCPWNuvQDB

The mask doesn't follow the `bunniesContainer`,  you need to set position on that Graphics object _or_ add it to the `bunniesContainer`.

Hope that helps clarify.

**Comment by ChristianKienle** [comment]
First of all thank you very much for your response.

> I'm not sure I understand this issue, it works as I would expect.

After reading your response and testing your hint (*adding the mask to the container*) I can say that the mask now works as I expected it to work initially.

> The mask doesn't follow the bunniesContainer, you need to set position on that Graphics object or add it to the bunniesContainer.

Yeah. Now everything seems to work again. So I will close this *issue* since it works as designed.

In my tests (that I did before creating this issue) I partially noticed what you wrote (that it works if the mask is re-positioned) – but I never thought that this is how it is supposed to work. But I am happy that my issue is *solved* and I learned something new.

And thank you very much for Pixi itself. I am using it to write a game and it really works well. I even use PIXI to render the game UI (with buttons, checkboxes, button groups, grid views, number inputs, …) since I liked it so much. :D

Keep up the fantastic work and thanks!
